swift_scality_backend/diskfile.py

```diff
@@ -253,14 +253,11 @@ class DiskFile(object):
 
     def __init__(self, client_collection, account, container, obj, use_splice,
                  logger):
-        # We quote separately each part and join parts with a plain forward
-        # slash. Note that `urllib.quote(_, '')` escapes forward slashes.
-        # Escaping `/` in addition to configuring Apache with
-        # `AllowEncodedSlashes NoDecode` makes sure that a Swift object named
-        # 'my//object' is treated by the backend differently than an object
-        # named 'my/object'.
-        self._name = '/'.join(urllib.quote(part, '')
-                              for part in (account, container, obj))
+        # We hash the account, container and object name so that no 'special'
+        # character will get in our way.
+        sha1 = hashlib.sha1()
+        map(sha1.update, [account, container, obj])
```

Comment:
Not very fond of this `map`. `map` is very useful when you construct a new value out of an old one, but shouldn't be used to generate side-effects.
Especially given Python3, where this construct wouldn't work at all!